jaxrs/src/main/java/com/cerner/beadledom/jaxrs/provider/ForwardedHeaderFilter.java

```diff
@@ -0,0 +1,72 @@
+import java.util.regex.Pattern;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * The ForwardedHeaderFilter reads the request context and determines if the client request was made
+ * from a secure protocol (HTTPS). If the request originated from secure context, it updates the
+ * request to reflect the original protocol.
+ *
+ * <p>To determine if the client request made in a secure context was forwarded by a load balancer
+ * or proxy, the "Forwarded" and "X-Forwarded-Proto" values are extracted and checked.</p>
+ *
+ * @author Nick Behrens
+ */
+@Provider
+@Priority(Priorities.HEADER_DECORATOR)
+@PreMatching
+public class ForwardedHeaderFilter implements ContainerRequestFilter {
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    String httpScheme = requestIsSecure(requestContext) ? "https" : "http";
+    requestContext.setRequestUri(
+        requestContext.getUriInfo().getRequestUriBuilder().scheme(httpScheme).build());
+  }
+
+  /**
+   * Checks if the request was made from a secure context or not.
+   * @param requestContext The {@link ContainerRequestContext} object containing the container request headers.
+   * @return true if the request was made from a secure context, false otherwise.
+   */
+  private boolean requestIsSecure(ContainerRequestContext requestContext) {
+    return requestContext.getSecurityContext().isSecure()
+        || hasSecureForwardedHeader(requestContext)
+        || hasSecureXForwardedProtoHeader(requestContext);
+  }
+
+  /**
+   * Checks the Forwarded header for a protocol value of https.
+   * @param requestContext The {@link ContainerRequestContext} object containing the container request headers.
+   * @return true if the `Forwarded` header is present and contains https; false otherwise.
+   */
+  private boolean hasSecureForwardedHeader(ContainerRequestContext requestContext) {
+    String forwardedHeader = requestContext.getHeaderString("Forwarded");
+    if (forwardedHeader == null) {
+      return false;
+    }
+    Pattern forwardedPairs = Pattern.compile("proto=(?<protocolValue>[^;]*)(;|\\z)");
+
+    Matcher matcher = forwardedPairs.matcher(forwardedHeader);
+
+    return matcher.find()
+        && matcher.group("protocolValue").contains("https");
+  }
+
+  /**
+   * Checks the X-Forwarded-Proto header for a value of https.
+   * @param requestContext The {@link ContainerRequestContext} object containing the container request headers.
+   * @return true if the `X-Forwarded-Proto` header is present and contains https; false otherwise.
+   */
+  private boolean hasSecureXForwardedProtoHeader(ContainerRequestContext requestContext) {
+    return "https".equals(requestContext.getHeaderString("X-Forwarded-Proto"));
```

Comment:
If the header is `X-Forwarded-Proto` is lower case (`x-forwarded-proto`) does this still work?

Reply:
It does, I verified it by setting the integration reverse proxy to use a lower case (x-forwarded-proto) header. I committed the changed nginx config: https://github.com/cerner/beadledom/pull/149/commits/f4695a52f8ff844164ee7e8b59263a32995bae32